server/desktop/src/main/java/dev/slimevr/desktop/platform/SteamVRBridge.kt

```diff
@@ -0,0 +1,426 @@
+		if (value == config.automaticSharedTrackersToggling) return
+
+		config.automaticSharedTrackersToggling = value
+		if (value) {
+			updateShareSettingsAutomatically()
+			RPCSettingsHandler.sendSteamVRUpdatedSettings(instance.protocolAPI, instance.protocolAPI.rpcHandler)
+		}
+		instance.configManager.saveConfig()
+	}
+
+	@VRServerThread
+	override fun changeShareSettings(role: TrackerRole?, share: Boolean) {
+		if (role == null) return
+		for (tr in shareableTrackers) {
+			if (tr.trackerPosition?.trackerRole == role) {
+				if (share) {
+					addSharedTracker(tr)
+				} else {
+					removeSharedTracker(tr)
+				}
+				config.setBridgeTrackerRole(role, share)
+				instance.configManager.saveConfig()
+			}
+		}
+	}
+
+	@VRServerThread
+	override fun createNewTracker(trackerAdded: TrackerAdded): Tracker {
+		val device = instance.deviceManager
+			.createDevice(
+				trackerAdded.trackerName,
+				trackerAdded.trackerSerial,
+				"OpenVR", // TODO : We need the manufacturer
+			)
+
+		val displayName: String
+		val needsReset: Boolean
+		if (trackerAdded.trackerId == 0) {
+			displayName = if (trackerAdded.trackerName == "HMD") {
+				"SteamVR Driver HMD"
+			} else {
+				"Feeder App HMD"
+			}
+			// TODO support needsReset = true for VTubing (GUI toggle?)
+			needsReset = false
+		} else {
+			displayName = trackerAdded.trackerName
+			needsReset = true
+		}
+
+		val tracker = Tracker(
+			device,
+			getNextLocalTrackerId(),
+			trackerAdded.trackerSerial,
+			displayName,
+			null,
+			trackerAdded.trackerId,
+			hasPosition = true,
+			hasRotation = true,
+			userEditable = true,
+			isComputed = true,
+			needsReset = needsReset,
+		)
+
+		device.trackers[0] = tracker
+		instance.deviceManager.addDevice(device)
+		val role = getById(trackerAdded.trackerRole)
+		if (role != null) {
+			tracker.trackerPosition = getByTrackerRole(role)
+		}
+		return tracker
+	}
+
+	// Battery Status
+	@VRServerThread
+	override fun writeBatteryUpdate(localTracker: Tracker) {
+		var lowestLevel = 200f // Arbitrarily higher than expected battery
+		// percentage
+		var trackerLevel = 0f // Tracker battery percentage on a scale from 0
+		// to 100. SteamVR expects a value from 0 to 1.
+		var trackerVoltage = 0f // Tracker voltage. This is used to determine
+		// if the tracker is being charged. owoTrack
+		// devices do not have a tracker voltage.
+		var isCharging = false
+
+		val allTrackers = instance.allTrackers
+		val role = localTracker.trackerPosition?.trackerRole ?: return
+
+		val batteryTrackers: MutableList<Tracker?> = FastList()
+
+		// batteryTrackers is filled with trackers that would give battery data
+		// for the SteamVR tracker according to its role. Warning: trackers
+		// inside batteryTrackers could be null, so there must be a null check
+		// when accessing its data.
+		batteryTrackers
+			.add(
+				getTrackerForSkeleton(
+					allTrackers,
+					getByTrackerRole(role) ?: return,
```

Comment:
I feel like I asked this before and there was a good reason, but why don't we just use `localTracker.trackerPosition` here?

Reply:
i sadly dont remember, and cant seem to find the reason why qwq

Reply:
Well uhhhhh I guess it's fine to just leave it that way lol, maybe put a comment to look into it or smth